cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/InMemoryDatasetFramework.java

```diff
@@ -614,6 +615,11 @@ public boolean hasType(String datasetTypeName) {
   }
 
   private void publishAudit(Id.DatasetInstance datasetInstance, AuditType auditType) {
+    // Don't publish audit for system datasets admin operations, there can be a deadlock
```

Comment:
IMO, system datasets should be published to the audit as well. If the dataset can be found via the tracker search, the audit logs should be published.